src/main/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,55 @@
+package stringcalculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+
+    private static final String SEPARATOR = ",|:";
+    private static final String CUSTOM_SEPARATOR_EXPRESSION = "//(.)\n(.*)";
+
+
+    public static String[] separator(String inputString) {
+        return inputString.split(SEPARATOR);
+    }
+
+    public static String[] customSeparator(String inputString) {
+        Matcher m = Pattern.compile(CUSTOM_SEPARATOR_EXPRESSION).matcher(inputString);
+        if (m.find()) {
+            String customDelimiter = m.group(1);
+            String[] tokens= m.group(2).split(customDelimiter);
+            return tokens;
+        }
+        return new String[]{inputString};
+    }
```

Comment:
역시 명사구로 메소드이름이 정해진 것이 조금 아쉬운 부분이에요.

또한 관점에 따라서는 `separator`와 `customSeparator`의 기능을 합쳐볼 수도 있지 않을까 하는 생각이 듭니다.

```suggestion
    public static String[] separator(String inputString) {
        Matcher m = Pattern.compile(CUSTOM_SEPARATOR_EXPRESSION).matcher(inputString);
        if (m.find()) {
            String customDelimiter = m.group(1);
            String[] tokens= m.group(2).split(customDelimiter + "|" + SEPARATOR);
            return tokens;
        }
        return inputString.split(SEPARATOR);
    }
```
물론 정답은 아닙니다! 위와 같은 코드도 고민의 한 흔적이 될 수 있을 것 같아요.

어려운 길(커스텀 구분자와 기본 구분자 모두를 적용하는 길)을 택하신만큼, 
더 많이 고민하고 더 많이 얻어갈 수 있는 단계가 되시길 응원합니다!